src/openvic-simulation/military/UnitInstance.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "openvic-simulation/map/Province.hpp"
+#include "openvic-simulation/military/Leader.hpp"
+#include "openvic-simulation/military/UnitType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include <string_view>
+#include <type_traits>
+#include <vector>
+
+namespace OpenVic {
+	template<typename T, typename std::enable_if<std::is_base_of<UnitType, T>::value>::type* = nullptr>
+	struct UnitInstance {
+	protected:
+		std::string PROPERTY_RW(unit_name);
+		T const& PROPERTY(unit_type); //can't change
+
+		fixed_point_t PROPERTY_RW(organisation);
+		fixed_point_t PROPERTY_RW(morale);
+		fixed_point_t PROPERTY_RW(strength);
+
+	protected:
+		UnitInstance(std::string_view new_unit_name, T const& new_unit_type) : 
+			unit_name { new_unit_name },
+			unit_type { new_unit_type }, 
+			organisation { new_unit_type.get_default_organisation() }, //TODO: modifiers
+			morale { 0 }, //TODO: where
+			strength { new_unit_type.get_max_strength() } {}
+	};
+
+	struct RegimentInstance : UnitInstance<RegimentType> {
+	private:
+		Pop const& PROPERTY(pop); // can't change
```

Comment:
```suggestion
		Pop& PROPERTY(pop); // can't change
```
While the pop it's referencing can't change, that pop's data will need to be changed through the reference to apply combat losses.